homeassistant/components/backup/config.py

```diff
@@ -176,12 +191,36 @@ def update(
         """Update config."""
         if agents is not UNDEFINED:
             for agent_id, agent_config in agents.items():
-                if agent_id not in self.data.agents:
-                    self.data.agents[agent_id] = AgentConfig(**agent_config)
+                agent_retention = agent_config.get("retention", UNDEFINED)
+                if agent_retention is UNDEFINED or agent_retention is None:
+                    new_agent_retention = None
                 else:
-                    self.data.agents[agent_id] = replace(
-                        self.data.agents[agent_id], **agent_config
+                    new_agent_retention = AgentRetentionConfig(
+                        copies=agent_retention.get("copies"),
+                        days=agent_retention.get("days"),
                     )
+                if agent_id not in self.data.agents:
+                    old_agent_retention = None
+                    self.data.agents[agent_id] = AgentConfig(
+                        protected=agent_config.get("protected", False),
+                        retention=new_agent_retention,
+                    )
+                else:
+                    new_agent_config = self.data.agents[agent_id]
+                    old_agent_retention = new_agent_config.retention
+                    if "protected" in agent_config:
+                        new_agent_config = replace(
+                            new_agent_config, protected=agent_config["protected"]
+                        )
+                    if "retention" in agent_config:
+                        new_agent_config = replace(
+                            new_agent_config, retention=new_agent_retention
```

Comment:
Yeah, we need to adjust the call here and check if agent_retention is UNDEFINED.

Reply:
Never mind, `if "retention" in agent_config` does that already.